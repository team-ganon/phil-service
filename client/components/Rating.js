import React from 'react';
import styles from '../CSS/rating.css';

const Rating = ({name, score}) => {
  let title = name.name;
  let style = {
    width: `${score * 2}0%`
  };
  return (
    <div className={styles.ratingDiv}>
      <p className={styles.title}>{title}</p>
      <div className={styles.ratingBarAndScore}>
        <div className={styles.outerLineDiv}>
          <div className={styles.innerLineDiv} style={style}></div>
        </div>
        <p className={styles.score}>{score}</p>
      </div>
    </div>
  );
};

export default Rating;