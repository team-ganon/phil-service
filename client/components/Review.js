import React from 'react';
import {format, parseISO} from 'date-fns';
import style from '../CSS/review.css';

const Review = ({review}) => {
  var result = format(
    parseISO(review.date),
    'MMMM yyyy'
  );
  return (
    <div className={style.reviewOuterDiv}>
      <div className={style.topOfReview}>
        <img className={style.image} height="48" width="48" src={review.userProfile.imageUrl} />
        <div className={style.nameDate}>
          <p>{review.userProfile.name}</p>
          <p className={style.date}>{result}</p>
        </div>
      </div>
      <p className={style.reviewParagraph}>{review.body}</p>
    </div>
  );
};

export default Review;