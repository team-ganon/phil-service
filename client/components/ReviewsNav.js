import React, { Component } from 'react';
import style from '../CSS/reviewnav.css';

class ReviewsNav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      buttons: 0,
      currentButtons: [],
      currentButton: 1,
      howManyLeft: this.props.reviews.length,
      below28: false
    };
  }

  componentDidMount() {
    if (this.props.reviews.length > 28) {
      this.setState({
        currentButtons: [1, 2, 3, 4]
      });
    } else {
      let numberOfButtons = Math.ceil(this.props.reviews.length / 7);
      let buttonArray = [];
      for (let i = 0; i < numberOfButtons; i++) {
        buttonArray.push(i + 1);
      }
      this.setState({
        currentButtons: buttonArray
      });
    }
  }

  changeForward() {
    this.setState({
      currentButton: this.state.currentButton += 1,
    });
    this.navLogic();
  }

  navLogic() {
    let lastReviews = this.props.reviews.slice(this.props.currentPlace);
    if (lastReviews.length < 28 && this.state.currentButton === 5) {
      let value = this.state.currentButtons[3];
      let remaining = Math.ceil(lastReviews.length / 7);
      let newButtons = [];
      for (let i = 1; i < remaining; i++) {
        newButtons.push(value + i);
      }
      this.setState({
        currentButtons: newButtons,
        currentButton: 1
      }); 
    } else if (this.state.currentButton === 5) {
      let value = this.state.currentButtons[3];
      var newButtons = [
        value + 1,
        value + 2,
        value + 3,
        value + 4
      ];
      this.setState({
        currentButtons: newButtons,
        currentButton: 1
      });
    }
  }

  changeBackword() {
    this.setState({
      currentButton: this.state.currentButton -= 1,
      howManyLeft: this.state.howManyLeft + 7
    });
    if (this.state.currentButton === 0) {
      let value = this.state.currentButtons[0];
      let newButtons = [
        value - 4,
        value - 3,
        value - 2,
        value - 1
      ];
      this.setState({
        currentButtons: newButtons,
        currentButton: 4
      });
    }
  }

  howManyLeft() {
    this.setState({
      howManyLeft: this.state.howManyLeft - 7
    });
  }

  render() {
    return (
      <div className={style.reviewsComponent}>
        {!(this.state.currentButtons[this.state.currentButton - 1] === 1) && 
        <button 
          className={style.backward}
          onClick={() => { this.changeBackword(); this.props.currentBackword(); }}>
          <ion-icon name="chevron-back-outline"></ion-icon>
        </button>}
        <div className={style.reviewNavDiv}>
          {this.state.currentButtons.map((num, i) => {
            if (this.state.currentButton === (i + 1)) {
              return <div key={i} className={style.currentButton}>{num}</div>;
            }
            return (
              <div key={i} className={style.reviewsButtons}>{num}</div>
            );
          })}
        </div>
        {!((this.props.reviews.length - this.props.currentPlace) <= 7) && <button 
          className={style.forward}
          onClick={() => { this.howManyLeft(); this.changeForward(); this.props.currentForward(); }}><ion-icon name="chevron-forward-outline"></ion-icon></button>}
      </div>
    );
  }
}

export default ReviewsNav;