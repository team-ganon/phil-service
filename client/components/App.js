import React, { Component } from 'react';
import reviewApiCall from './api';
import style from '../CSS/app.css';
import ReviewsNav from './ReviewsNav';
import ReviewsHeader from './ReviewsHeader';
import Review from './Review';
import Rating from './Rating';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: [],
      currentPlace: 0,
      currentReviews: [],
      total: 0,
      average: 0,
      error: false,
      cleanliness: 0,
      communication: 0,
      value: 0,
      accuracy: 0,
      checkIn: 0,
      location: 0
    };

    this.currentForward = () => {
      let current = this.state.reviews.slice(this.state.currentPlace + 7, this.state.currentPlace + 14);
      this.setState({
        currentPlace: this.state.currentPlace + 7,
        currentReviews: current
      });
    };

    this.currentBackword = () => {
      let current = this.state.reviews.slice(this.state.currentPlace - 7, this.state.currentPlace);
      this.setState({
        currentPlace: this.state.currentPlace - 7,
        currentReviews: current
      });
    };
  }

  componentDidMount() {
    let endPoint = window.location.href.split('=');
    reviewApiCall(endPoint[1])
      .then(result => console.log(result))
      .catch(e => this.setState(e));
  }

  render() {
    return (
      <div className={style.app}>
        <ReviewsHeader 
          average={this.state.average}
          total={this.state.total}
        />
        <div className={style.ratingHolderDiv}>
          <Rating name={{name: 'Cleanliness'}}score={this.state.cleanliness}/>
          <Rating name={{name: 'Communication'}}score={this.state.communication}/>
          <Rating name={{name: 'Value'}}score={this.state.value}/>
          <Rating name={{name: 'Accuracy'}}score={this.state.accuracy}/>
          <Rating name={{name: 'Check-in'}}score={this.state.checkIn}/>
          <Rating name={{name: 'Location'}}score={this.state.location}/>
        </div>
        {this.state.currentReviews.map((review, i) => <Review review={review} key={i} />)}
        {this.state.reviews.length > 0 && 
          <ReviewsNav 
            currentPlace={this.state.currentPlace}
            currentForward={this.currentForward}
            currentBackword={this.currentBackword}
            reviews={this.state.reviews}
          />
        }
      </div>
    );
  }
}

export default App;