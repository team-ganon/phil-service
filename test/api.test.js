import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/components/App';
import Review from '../client/components/Review.js';
import ReviewHeader from '../client/components/ReviewsHeader';
import ReviewsNav from '../client/components/ReviewsNav';
import Rating from '../client/components/Rating';

let reviewSample = {
  '_id': '5e718b9cce115a5271351577',
  'user': '5e718b9bce115a5271351513',
  'rental': 1,
  'body': 'Cumque pariatur quisquam quibusdam non qui qui ratione. Autem ratione rerum ut id rerum voluptatem adipisci iure accusantium. Ipsum voluptas maiores occaecati quaerat est corporis omnis perspiciatis.',
  'cleanliness': 4,
  'communication': 4,
  'value': 1,
  'accuracy': 1,
  'checkIn': 4,
  'location': 4,
  'date': '2020-03-18T02:46:52.298Z',
  '__v': 0,
  'userProfile': {
    '_id': '5e718b9bce115a5271351513',
    'name': 'David Purdy',
    'imageUrl': 'https://hackreactoramazonfrontendcapstone.s3-us-west-2.amazonaws.com/1.jpeg',
    '__v': 0
  }
};

describe('Testing /api/rentals/:id', () => {

  it('Testing /api/rentals/:id', () => {
    return fetch(`http://localhost:3001/api/rentals/1`)
      .then(data => data.json())
      .then(data => expect(data[1].userProfile.name).toEqual('Rosalia Crona'))
      .catch((e) => console.log(e));
  });

  it('<App /> currentPlace state should be zero', () => {
    const wrapper = shallow(<App />);
    const currentPlaceState = wrapper.state().currentPlace;
    expect(currentPlaceState).toEqual(0);
  });

});

describe('App', () => {

  it('Review Component', () => {
    const ReviewJson = renderer.create(<Review review={reviewSample} />).toJSON();
    expect(ReviewJson).toMatchSnapshot();
  });

  it('ReviewHeader Component', () => {
    const ReviewHeaderJson = renderer.create(<ReviewHeader average={4} total={100} />).toJSON();
    expect(ReviewHeaderJson).toMatchSnapshot();
  });

  it('ReviewsNav Component', () => {
    const ReviewsNavJson = renderer.create(<ReviewsNav currentPlace={0} reviews={20} />).toJSON();
    expect(ReviewsNavJson).toMatchSnapshot();
  });

  it('Rating Component', () => {
    const RatingJson = renderer.create(<Rating name={{name: 'Cleanliness'}} score={3} />).toJSON();
    expect(RatingJson).toMatchSnapshot();
  });
  
});

