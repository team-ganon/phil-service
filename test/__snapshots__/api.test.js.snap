// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rating Component 1`] = `
<div
  className="ratingDiv"
>
  <p
    className="title"
  >
    Cleanliness
  </p>
  <div
    className="ratingBarAndScore"
  >
    <div
      className="outerLineDiv"
    >
      <div
        className="innerLineDiv"
        style={
          Object {
            "width": "60%",
          }
        }
      />
    </div>
    <p
      className="score"
    >
      3
    </p>
  </div>
</div>
`;

exports[`App Review Component 1`] = `
<div
  className="reviewOuterDiv"
>
  <div
    className="topOfReview"
  >
    <img
      className="image"
      height="48"
      src="https://hackreactoramazonfrontendcapstone.s3-us-west-2.amazonaws.com/1.jpeg"
      width="48"
    />
    <div
      className="nameDate"
    >
      <p>
        David Purdy
      </p>
      <p
        className="date"
      >
        March 2020
      </p>
    </div>
  </div>
  <p
    className="reviewParagraph"
  >
    Cumque pariatur quisquam quibusdam non qui qui ratione. Autem ratione rerum ut id rerum voluptatem adipisci iure accusantium. Ipsum voluptas maiores occaecati quaerat est corporis omnis perspiciatis.
  </p>
</div>
`;

exports[`App ReviewHeader Component 1`] = `
<div
  className="reviewHeaderDiv"
>
  <h2
    className="reviewsHeaderH2"
  >
    Reviews
  </h2>
  <div
    className="bottomDivOfHeader"
  >
    <div
      className="containerForStarReviews"
    >
      <div
        className="star"
      >
        <ion-icon
          name="star"
        />
        <p
          className="averageTotal"
        >
          4
        </p>
      </div>
      <p
        className="average"
      >
        <span
          className="reviewCount"
        >
          100
        </span>
         reviews
      </p>
    </div>
    <div
      className="search"
    >
      <input
        placeholder="Search Reviews"
        type="text"
      />
      <ion-icon
        name="search-outline"
      />
    </div>
  </div>
</div>
`;

exports[`App ReviewsNav Component 1`] = `
<div
  className="reviewsComponent"
>
  <button
    className="backward"
    onClick={[Function]}
  >
    <ion-icon
      name="chevron-back-outline"
    />
  </button>
  <div
    className="reviewNavDiv"
  />
  <button
    className="forward"
    onClick={[Function]}
  >
    <ion-icon
      name="chevron-forward-outline"
    />
  </button>
</div>
`;
